src/custom-element.js

```diff
@@ -934,32 +934,23 @@ function createBaseCustomElementClass(win) {
         this.resources_ = resourcesForDoc(this.ampdoc_);
       }
       if (!this.everAttached) {
+        this.everAttached = true;
+
+        try {
+          this.layout_ = applyLayout_(this);
+        } catch (e) {
+          reportError(e, this);
+        }
         if (!isStub(this.implementation_)) {
           this.tryUpgrade_();
         }
         if (!this.isUpgraded()) {
           this.classList.add('amp-unresolved');
           this.classList.add('i-amphtml-unresolved');
+          // amp:attached is dispatched from the ElementStub class when it
```

Comment:
I'm copying the exact comment from https://github.com/ampproject/amphtml/pull/8229/files/d4690520ef30a0b5810fa207f85d27bcd48e3ec3#diff-8e1cdba946c07a7e631ef74645a5d6f0L950.